modules/monitoring-troubleshooting-alertmanager-configurations.adoc

```diff
@@ -0,0 +1,188 @@
+// Module included in the following assemblies:
+//
+// * monitoring/troubleshooting-monitoring-issues.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="troubleshooting-alertmanager-configurations_{context}"]
+= Troubleshooting Alertmanager configuration
+
+If your Alertmanager configuration does not work properly, you can compare the `alertmanager-main` secret with the running Alertmanager configuration to identify possible errors. You can also test your alert routing configuration by creating a test alert.
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` cluster role.
+* You have installed the {oc-first}.
+
+.Procedure
+
+. Compare the `alertmanager-main` secret with the running Alertmanager configuration:
+
+.. Extract the Alertmanager configuration from the `alertmanager-main` secret into the `alertmanager.yaml` file:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring get secret alertmanager-main --template='{{ index .data "alertmanager.yaml" }}' | base64 --decode > alertmanager.yaml 
+----
+
+.. Pull the running Alertmanager configuration from the API:
++
+[source,terminal]
+----
+$ oc exec -n openshift-monitoring alertmanager-main-0 -- amtool config show --alertmanager.url http://localhost:9093
+----
++
+.Example output
+[source,terminal]
+----
+global:
+  resolve_timeout: 5m
+  http_config:
+    follow_redirects: true
+    enable_http2: true
+    proxy_from_environment: true
+...
+route:
+  receiver: default
+  group_by:
+  - namespace
+  continue: false
+  routes:
+  ...
+  - matchers: # <1>
+    - service="example-app"
+    continue: false
+    routes:
+    - receiver: team-frontend-page
+      matchers:
+      - severity="critical"
+      continue: false
+  ...
+receivers:
+...
+- name: team-frontend-page # <2>
+  pagerduty_configs:
+  - send_resolved: true
+    http_config:
+      authorization:
+        type: Bearer
+        credentials: <secret>
+      follow_redirects: true
+      enable_http2: true
+      proxy_from_environment: true
+    service_key: <secret>
+    url: https://events.pagerduty.com/v2/enqueue
+    ...
+templates: []
+----
+<1> The example shows the route to the `team-frontend-page` receiver. Alertmanager routes alerts with `service="example-app"` and `severity="critical"` labels to this receiver.
+<2> The `team-frontend-page` receiver configuration. The example shows PagerDuty as a receiver.
+
+.. Compare the contents of the `route` and `receiver` fields of the `alertmanager.yaml` file with the fields in the running Alertmanager configuration. Look for any discrepancies.
```

Comment:
comparing could be tedious, maybe we should think about a diff command or sth.
If we only suspect the secret to not be taken into account by AM, maybe checking a log somewhere after changing the secret is sufficient?